### ISSUE TITLE
#include is missed if line starts with any whitespace

### ISSUE THREAD

**Issue by coder-arn** [fastbuild/fastbuild#56]
In the example below FileA.hpp is not added as a dependency and modifications to FileA.hpp don't trigger a rebuild of main.cpp.

```
main.cpp

    #include "FileA.hpp"    // Missed since line starts with whitespace
#  include "FileB.hpp"      // Correctly added as a dependency
#include "FileC.hpp"        // Correctly added as a dependency
```

This is especially problematic for us as we include Bullet in our project and it uses tabs before #include all over the place.

P.S.  Thank you for making fastbuild, its awesome!!

**Comment by ffulin** [comment]
Hello.  FASTBuild doesn't parse the includes directly - it uses the output of the compiler (usually the preprocessor, but sometimes /showincludes).  Maybe with some compilers the format of the output is altered when the include is indented as you've shown, which perhaps FASTBuild fails to parse?  Pretty weird, but certainly possible.

Can you tell me which compiler you are using?

**Comment by ffulin** [comment]
I've reproduced the issue with the MCVC compiler.  When using -E, the first line is inconsistently output in the preprocessed output.
```C++
        #line 1 "c:\\test\\code\\FileA.hpp"
#line 1 "c:\\test\\code\\FileB.hpp"
#line 1 "c:\\test\\code\\FileC.hpp"
```
Is this the compiler you are using?

**Comment by kylawl** [comment]
Ya we're using MSVC

**Comment by ffulin** [comment]
There was an (incorrect) assumption in the parsing of the pre-processed output for the MS compiler that assumed the #line directive always started at the beginning of the line. I have fixed this bug and am working on updating the test coverage to catch this case and to be sure there are no regressions.

The bug fix will be in the next version (v0.87). I'll update this issue with more info soon.

Thanks for reporting the bug!

**Comment by kylawl** [comment]
Cool, thanks ffulin

**Comment by coder-arn** [comment]
Thank you from me as well!

**Comment by ffulin** [comment]
A fix has been commited to the Perforce mainline. It'll be in the next release (v0.87). I've also [submitted the fix to "dev"](https://github.com/fastbuild/fastbuild/commit/e7356c8eea7eb0037e69a7ed2e74710686a4acf7) if anyone needs it urgently.

As a bonus, the new include extraction code is also about twice as fast (only saves a few ms per large object, but it's still a bonus)

I'll keep this issue open until the new version is released.

**Comment by ffulin** [comment]
The fix is now live in the v0.87 release.

**Comment by kylawl** [comment]
Awesome, thanks!
